### PR TITLE
dotCMS/core#13554 Handle template and layout issues

### PR DESCRIPTION
**Pull request by fmontes**
- Refactor render HTML endpoint
- Create new big interface with all the page data
- No more 2 endpoints for page and layout

### CODE REVIEW

File: src/app/portlets/dot-edit-page/content/shared/iframe-edit-mode.css.ts [hunk 1]
```diff
@@ -139,14 +139,19 @@ export const EDIT_PAGE_CSS = `
         opacity: 1;
     }
 
-     .dotedit-contentlet__toolbar button {
+    .dotedit-contentlet__toolbar button {
         margin-right: 8px;
```
Comment (hmoreras) [inline src/app/portlets/dot-edit-page/content/shared/iframe-edit-mode.css.ts]: create variable equivalent to `$basic-padding` in this file ?

Comment (fmontes) [reply src/app/portlets/dot-edit-page/content/shared/iframe-edit-mode.css.ts]: We should.

File: src/app/api/services/dot-dialog/dot-dialog.service.spec.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ describe('DotDialogService', () => {
 
     it('should clear alert model', () => {
         service.alert(mockData);
-        service.clearAlert();
+        service.alertAccept(new MouseEvent('click'));
         expect(service.alertModel).toEqual(null);
     });
```
Comment (freddyucv) [inline src/app/api/services/dot-dialog/dot-dialog.service.spec.ts]: I think we are missing a test to check the alertReject method

File: src/app/portlets/dot-edit-page/content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts [hunk 3]
```diff
@@ -204,7 +203,7 @@ export class DotEditPageToolbarComponent implements OnInit, OnChanges {
                 label: this.dotMessageService.get('editpage.toolbar.edit.page'),
                 value: PageMode.EDIT,
                 styleClass:
-                    !pageState.page.canEdit || pageState.state.lockedByAnotherUser ? 'edit-page-toolbar__state-selector-item--disabled' : ''
+                    !pageState.page.canEdit || !pageState.page.canLock ? 'edit-page-toolbar__state-selector-item--disabled' : ''
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/components/dot-edit-page-toolbar/dot-edit-page-toolbar.component.ts]: at least one test should be updated with this change

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-resolver.service.ts [hunk 4]
```diff
@@ -18,18 +22,44 @@ import { DotPageStateService } from './dot-page-state/dot-page-state.service';
...
-        return this.dotPageStateService.get(route.queryParams.url).catch((err: ResponseView) => {
-            this.dotHttpErrorManagerService.handle(err).subscribe((res: any) => {
-                if (!res.redirected) {
-                    this.dotRouterService.gotoPortlet('/c/site-browser');
+        return this.dotPageStateService
+            .get(route.queryParams.url)
+            .map((dotRenderedPageState: DotRenderedPageState) => {
+                const currentSection = route.children[0].url[0].path;
+                const isLayout = currentSection === 'layout';
+                const userCantEditLayout = isLayout && !dotRenderedPageState.page.canEdit;
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-resolver.service.ts]: maybe we chould put a TODO here_: //TODO look for better way to do this

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-resolver.service.ts [hunk 5]
```diff
@@ -18,18 +22,44 @@ import { DotPageStateService } from './dot-page-state/dot-page-state.service';
...
+
+    private errorHandler(err: ResponseView): Observable<DotRenderedPageState> {
+        this.dotHttpErrorManagerService.handle(err).subscribe((res: any) => {
+            if (!res.redirected) {
+                this.dotRouterService.gotoPortlet('/c/site-browser');
+            }
         });
+        return Observable.of(null);
     }
 }
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-resolver.service.ts]: What happend with this class's test?

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts [hunk 6]
```diff
@@ -62,8 +63,10 @@ export class DotEditLayoutDesignerComponent implements OnInit {
 
         this.setupLayout();
 
-        if (!this.isLayout()) {
+        if (!this.isLayout() && this.pageState.template.canEdit) {
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts]: I think we need so me more test when the dialog must to be showed or not.
I think we have only have 'should show select edit layout/template dialog'

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts [hunk 7]
```diff
@@ -125,13 +128,13 @@ export class DotEditLayoutDesignerComponent implements OnInit {
...
             () => {
                 // TODO: This extra request will change once the this.pageViewService.save return a DotPageView object.
-                this.pageViewService.get(this.route.snapshot.queryParams.url).subscribe((pageView: DotPageView) => {
-                    this.dotGlobalMessageService.display(this.dotMessageService.get('dot.common.message.saved'));
-                    this.setupLayout(pageView);
-                });
+                // this.pageViewService.get(this.route.snapshot.queryParams.url).subscribe((pageView: DotPageView) => {
+                //     this.dotGlobalMessageService.display(this.dotMessageService.get('dot.common.message.saved'));
+                //     this.setupLayout(pageView);
+                // });
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts]: we dont should push commented code

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts [hunk 8]
```diff
@@ -185,18 +190,39 @@ export class DotEditLayoutDesignerComponent implements OnInit {
...
+                        accept: this.dotMessageService.get('editpage.layout.dialog.edit.page'),
+                        reject: this.dotMessageService.get('editpage.layout.dialog.edit.template')
+                    },
+                    accept: () => {
+                        this.setEditLayoutMode();
+                    }
+                });
+            });
+
+        // this.showTemplateLayoutSelectionDialog = true;
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts]: no commented code

**Pull request merged**